File: app/src/main/java/com/blankj/androidutilcode/feature/core/process/ProcessActivity.java
package com.blankj.androidutilcode.feature.core.process;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import com.blankj.androidutilcode.R;
import com.blankj.androidutilcode.base.BaseBackActivity;
import com.blankj.utilcode.util.ProcessUtils;
import com.blankj.utilcode.util.SpanUtils;

import java.util.Iterator;
import java.util.Set;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/10/13
 *     desc  : demo about ProcessUtils
 * </pre>
 */
public class ProcessActivity extends BaseBackActivity {

    public static void start(Context context) {
        Intent starter = new Intent(context, ProcessActivity.class);
        context.startActivity(starter);
    }

    private TextView tvAboutProcess;

    @Override
    public void initData(@Nullable Bundle bundle) {

    }

    @Override
    public int bindLayout() {
        return R.layout.activity_process;
    }

    @Override
    public void initView(Bundle savedInstanceState, View contentView) {
        getToolBar().setTitle(getString(R.string.demo_process));

        findViewById(R.id.btn_kill_all_background_processes).setOnClickListener(this);
        tvAboutProcess = findViewById(R.id.tv_about_process);
        Set<String> set = ProcessUtils.getAllBackgroundProcesses();
        tvAboutProcess.setText(new SpanUtils()
                .appendLine("getForegroundProcessName: " + ProcessUtils.getForegroundProcessName())
                .appendLine("getAllBackgroundProcesses: " + getSetItems(set))
                .appendLine("size: " + set.size())
                .appendLine("isMainProcess: " + ProcessUtils.isMainProcess())
                .append("getCurrentProcessName: " + ProcessUtils.getCurrentProcessName())
                .create()
        );
    }

    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {
        switch (view.getId()) {
            case R.id.btn_kill_all_background_processes:
                Set<String> set = ProcessUtils.getAllBackgroundProcesses();
                Set<String> set1 = ProcessUtils.killAllBackgroundProcesses();
                tvAboutProcess.setText(new SpanUtils()
                        .appendLine("getForegroundProcessName: " + ProcessUtils.getForegroundProcessName())
                        .appendLine("getAllBackgroundProcesses: " + getSetItems(set))
                        .appendLine("size: " + set.size())
                        .appendLine("killAllBackgroundProcesses: " + getSetItems(set1))
                        .appendLine("size: " + set1.size())
                        .appendLine("isMainProcess: " + ProcessUtils.isMainProcess())
                        .append("getCurrentProcessName: " + ProcessUtils.getCurrentProcessName())
                        .create()
                );
                break;
        }
    }

    private String getSetItems(Set<String> set) {
        Iterator<String> iterator = set.iterator();
        StringBuilder sb = new StringBuilder();
        while (iterator.hasNext()) {
            sb.append(iterator.next()).append("\n");
        }
        return sb.toString();
    }
}

File: feature/utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/process/ProcessActivity.kt
package com.blankj.utilcode.pkg.feature.process

import android.content.Context
import android.content.Intent
import com.blankj.common.activity.CommonActivity
import com.blankj.common.item.CommonItem
import com.blankj.common.item.CommonItemClick
import com.blankj.common.item.CommonItemTitle
import com.blankj.utilcode.pkg.R
import com.blankj.utilcode.util.CollectionUtils
import com.blankj.utilcode.util.ProcessUtils

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/10/13
 * desc  : demo about ProcessUtils
 * ```
 */
class ProcessActivity : CommonActivity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, ProcessActivity::class.java)
            context.startActivity(starter)
        }
    }

    override fun bindTitleRes(): Int {
        return R.string.demo_process
    }

    override fun bindItems(): MutableList<CommonItem<*>> {
        val set = ProcessUtils.getAllBackgroundProcesses()
        return CollectionUtils.newArrayList(
                CommonItemTitle("getForegroundProcessName", ProcessUtils.getForegroundProcessName()!!),
                CommonItemTitle("getAllBackgroundProcesses -> ${set.size}", getSetItems(set)),
                CommonItemTitle("isMainProcess", ProcessUtils.isMainProcess().toString()),
                CommonItemTitle("getCurrentProcessName", ProcessUtils.getCurrentProcessName()),

                CommonItemClick(R.string.process_kill_all_background).setOnItemClickListener { _, item, _ ->
                    val bgSet = ProcessUtils.getAllBackgroundProcesses()
                    val killedSet = ProcessUtils.killAllBackgroundProcesses()
                    itemsView.updateItems(
                            CollectionUtils.newArrayList(
                                    CommonItemTitle("getForegroundProcessName", ProcessUtils.getForegroundProcessName()),
                                    CommonItemTitle("getAllBackgroundProcesses -> ${bgSet.size}", getSetItems(bgSet)),
                                    CommonItemTitle("killAllBackgroundProcesses -> ${killedSet.size}", getSetItems(killedSet)),
                                    CommonItemTitle("isMainProcess", ProcessUtils.isMainProcess().toString()),
                                    CommonItemTitle("getCurrentProcessName", ProcessUtils.getCurrentProcessName()),
                                    item
                            )
                    )
                }
        )
    }

    private fun getSetItems(set: Set<String>): String {
        val iterator = set.iterator()
        val sb = StringBuilder()
        while (iterator.hasNext()) {
            sb.append("\n").append(iterator.next())
        }
        return if (sb.isNotEmpty()) sb.deleteCharAt(0).toString() else ""
    }
}
